public/algorytm.php

```diff
@@ -1,6 +1,136 @@
-<?php 
-/*
-	Algorytm przetwarzania obrazu ze skali szarości na kolor z wykorzystaniem mapowania pikseli.
-*/
+<?php
+/**
+ * createImage()	- Tworzy nowy obraz z pliku
+ * @param string	$filename nazwa pliku
+ * 
+ * @return zwraca obraz lub w przypadu niepowodzenia false;
+ */
+function createImage($filename){
+		switch (exif_imagetype($filename)) {
+			case IMAGETYPE_GIF:
+				return imagecreatefromgif($filename);
+				break;
+			case IMAGETYPE_JPEG:
+				return imagecreatefromjpeg($filename);
+				break;
+			case IMAGETYPE_PNG:
+				return imagecreatefrompng($filename);
+				break;
+			case IMAGETYPE_BMP:
+				return imagecreatefrombmp($filename);
+				break;
+			default:
+				exit("Unsupported file type ($filename)");
```

Comment:
Ogólnie używanie exita() jest brzydkie w PHPie, ale zostawmy go tutaj. Jeśli będziemy chcieli pisać testy, to wtedy to zrefaktoryzujemy